frontend/src/components/DefenceBox/DefenceBox.tsx

```diff
@@ -21,13 +22,25 @@ function DefenceBox({
 		config: DefenceConfigItem[]
 	) => Promise<boolean>;
 }) {
+	function getPromptEnclosureDefences() {
+		return defences.filter((defence) => {
+			return PROMPT_ENCLOSURE_DEFENCES.some((id) => id === defence.id);
+		});
+	}
+	function getNotPromptEnclosureDefences() {
+		return defences.filter((defence) => {
+			return !PROMPT_ENCLOSURE_DEFENCES.some((id) => id === defence.id);
+		});
+	}
+
 	return (
 		<div className="defence-box">
-			{defences.map((defence, index) => {
+			{getNotPromptEnclosureDefences().map((defence, index) => {
 				return (
 					<DefenceMechanism
 						key={index}
 						defenceDetail={defence}
+						promptEnclosureDefences={getPromptEnclosureDefences()}
```

Comment:
DefenceMechanism was nice and generic previously; now it knows specifically about prompt enclosure defences.

I think this needs a cleaner solution, I'll have a think.

Reply:
Added #745 for this.